src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -416,10 +416,28 @@ public override async Task FinalizeCheckpointCompletedAsync(Guid guid)
                     this.fht.Log.ShiftBeginAddress(shiftBeginAddress.Value);
                 }
 
+                long versionBeforeCheckpoint = this.mainSession.Version;
+
```

Comment:
Just for my understanding: here the `session` refers to what in FASTER? Is it an ongoing process, the state before checkpointing in storage, or something else? I'd appreciate having that as an inline comment in the code for future reference.